### PR TITLE
[chore] add metadata status checks on the distribution

### PR DESCRIPTION
**Pull request by atoulme**
Add a test checking that the status metadata of a component matches their presence in the otelcontribcol test suite.

### CODE REVIEW

**Comment by atoulme** [comment]
This fails because are detecting a number of components are not in the test suite when they should be.

File: cmd/otelcontribcol/distribution_metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			require.NoError(tt, err)
+			if m.Status == nil {
+				tt.Skip("no status present, skipping", metadataComponent)
+				return
+			}
+			if m.Status.Stability == "development" {
+				tt.Skip("component in development, skipping", metadataComponent)
+				return
+			}
+			fmt.Println(m.Type)
```
Comment (dmitryax) [inline cmd/otelcontribcol/distribution_metadata_test.go:55]: debug

**Review by dmitryax — APPROVED** [review]

File: cmd/otelcontribcol/distribution_metadata_test.go (hunk 1 above)
Comment (atoulme) [reply cmd/otelcontribcol/distribution_metadata_test.go]: fixed, thanks!

**Review by codeboten — COMMENTED** [review]
Awesome, this is a great addition. It would be great to also check the manifest file in the releases repo for the specific distro, though i'm not sure what the best way to do this would be.

How do you want to proceed forward to get the check passing? Should this PR wait until all the missing components are added to the tests?

**Comment by atoulme** [comment]
I went into this without a clear idea of how many corrections I would need to apply. Now that I have an idea of it, I can offer a few paths forward:

one is that we merge the tests but skipped for now. We set a comment to turn them on soon.
Another is to open a few PRs with the corrections I have so far.

**Comment by dmitryax** [comment]
> Another is to open a few PRs with the corrections I have so far.

I'd vote for this one

**Comment by atoulme** [comment]
Definitely. I'll add those and keep rebasing this PR over time to keep us closer to the tests passing.

**Comment by mx-psi** [comment]
> Definitely. I'll add those and keep rebasing this PR over time to keep us closer to the tests passing.

Maybe it's easier to have an allowlist for components where we will ignore the test failures and merge the PR now? That way we avoid introducing errors on components that are currently passing

**Comment by atoulme** [comment]
It's really close, just need a tidy

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Seems like it's still failing

**Comment by atoulme** [comment]
More components have metadata, so this test now detects more components missing from otelcontribcol:
```
--- FAIL: TestComponentsArePresent (0.73s)
    --- FAIL: TestComponentsArePresent/../../cmd/mdatagen/metadata.yaml (0.00s)
        distribution_metadata_test.go:60: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/otelcontribcol/distribution_metadata_test.go:60
            	Error:      	Expected value not to be nil.
            	Test:       	TestComponentsArePresent/../../cmd/mdatagen/metadata.yaml
            	Messages:   	missing receiver: test
    --- FAIL: TestComponentsArePresent/../../extension/storage/dbstorage/metadata.yaml (0.00s)
        distribution_metadata_test.go:[56](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5095304215/jobs/9160133239?pr=21275#step:8:57): 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/otelcontribcol/distribution_metadata_test.go:56
            	Error:      	Expected value not to be nil.
            	Test:       	TestComponentsArePresent/../../extension/storage/dbstorage/metadata.yaml
            	Messages:   	missing extension: db_storage
    --- FAIL: TestComponentsArePresent/../../receiver/statsdreceiver/metadata.yaml (0.00s)
        distribution_metadata_test.go:[60](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5095304215/jobs/9160133239?pr=21275#step:8:61): 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/otelcontribcol/distribution_metadata_test.go:60
            	Error:      	Expected value not to be nil.
            	Test:       	TestComponentsArePresent/../../receiver/statsdreceiver/metadata.yaml
            	Messages:   	missing receiver: statsdreceiver
```

I'll add them.

**Review by codeboten — COMMENTED** [review]
Some conflicts to resolve

**Review by dmitryax — APPROVED** [review]

**Pull request merged**